### DISCUSSION TITLE
After sealing, the file is transferred to HDD. The designated worker works like AP PC1 PC2 tasks enable/disable emove

### DISCUSSION THREAD

**Discussion by bxs17** [filecoin-project/lotus#8936] [General Q&A] [answered]
### Checklist

- [X] This is **not** a new feature or an enhancement to the Filecoin protocol. If it is,  please open an [FIP issue](https://github.com/filecoin-project/FIPs/blob/master/FIPS/fip-0001.md).
- [X] This is **not** brainstorming ideas. If you have an idea you'd like to discuss, please open a new discussion on [the lotus forum](https://github.com/filecoin-project/lotus/discussions/categories/ideas) and select the category as `Ideas`.
- [X] I **have** a specific, actionable, and well motivated feature request to propose.

### Lotus component

- [ ] lotus daemon - chain sync
- [ ] lotus miner - mining and block production
- [X] lotus miner/worker - sealing
- [ ] lotus miner - proving(WindowPoSt)
- [X] lotus miner/market - storage deal
- [ ] lotus miner/market - retrieval deal
- [ ] lotus miner/market - data transfer
- [ ] lotus client
- [ ] lotus JSON-RPC API
- [ ] lotus message management (mpool)
- [ ] Other

### What is the motivation behind this feature request? Is your feature request related to a problem? Please describe.

After sealing, the file is transferred to HDD. The designated worker works like AP PC1 PC2 tasks enable/disable emove
Generally, it is completed by the worker running on the HDD server Reduce network overhead
You can also turn off miner's move

### Describe the solution you'd like

The move task provides control commands and configurations

**Comment by rjan90** [comment]
Hey @bxs17! I´m not sure I understand the question, could you maybe expand a bit more about what you are wondering about/what issue you are facing? 😄

**Comment by bxs17** [comment]
The seal worker does not have persistent storage attached. After sealing, miner gets to persistent storage. You want to specify the worker to perform the get move operation. There is a switch to set whether miner opens the get move task。tasks type is？fget？




---Original---
From: ***@***.***&gt;
Date: Tue, Jun 28, 2022 13:51 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [filecoin-project/lotus] After sealing, the file is transferredto HDD. The designated worker works like AP PC1 PC2 tasks enable/disable emove(Discussion #8936)




 
Hey @bxs17! I´m not sure I understand the question, could you maybe expand a bit more about what you are wondering about/what issue you are facing? 😄

**Reply by rjan90** [reply]
Okay, so if I understand correctly, you want a worker to move the data to the long term storage?

If so one option is to just run lotus-worker on storage boxes with all compute tasks disabled; this way all storage accesses are going through independent tcp streams, and straight to whatever filesystem. But that requires:
- Post workers for reading challenges from remote storage: https://lotus.filecoin.io/storage-providers/seal-workers/post-workers/

To run a worker on the a storage box, just so it schedules the moving of sectors, just disable all tasks with: `lotus-worker run --no-default`

**Comment by bxs17** [comment]
Tried. But miner can also do transfer tasks. The task was not given to the worker. Whether miner can be closed
W？



---Original---
From: ***@***.***&gt;
Date: Tue, Jun 28, 2022 14:51 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [filecoin-project/lotus] After sealing, the file is transferredto HDD. The designated worker works like AP PC1 PC2 tasks enable/disable emove(Discussion #8936)




 
Okay, so if I understand correctly, you want a worker to move the data to the long term storage?
 
If so one option is to just run lotus-worker on storage boxes with all compute tasks disabled; this way all storage accesses are going through independent tcp streams, and straight to whatever filesystem. But that requires:
  
Post workers for reading challenges from remote storage: https://lotus.filecoin.io/storage-providers/seal-workers/post-workers/ 
 
To run a worker on the a storage box, just so it schedules the moving of sectors, just disable all tasks with: lotus-worker run --no-default

**Comment by rjan90** [comment, marked as answer]
> But miner can also do transfer tasks.

That is probably because your lotus-miner still has the paths to the long-term storage. The scenario I outlined above, is where the lotus-miner is only doing the scheduling, and has windowPoSt-workers attached, and therefore does not need the paths to the long-term storage on it.

In your case it sound like the lotus-miner still has storage-paths set, so if you want to move to an architecture where only the workers does the transfers task you would need to remove them. As a general advice here, please test out this setup first before chaning it on your production. And also keep a backup of the storage.json file.

**Comment by bxs17** [comment]
Or in other words, how to improve the priority of worker move tasks。

lotus-worker run --no-default&nbsp;

Can it be achieved


---Original---
From: ***@***.***&gt;
Date: Tue, Jun 28, 2022 15:34 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [filecoin-project/lotus] After sealing, the file is transferredto HDD. The designated worker works like AP PC1 PC2 tasks enable/disable emove(Discussion #8936)




  
But miner can also do transfer tasks.
  
That is probably because your lotus-miner still has the paths to the long-term storage. The scenario I outlined above, is where the lotus-miner is only doing the scheduling, and has windowPoSt-workers attached, and therefore does not need the paths to the long-term storage on it.
 
In your case it sound like the lotus-miner still has storage-paths set, so if you want to move to an architecture where only the workers does the transfers task you would need to remove them. As a general advice here, please test out this setup first before chaning it on your production. And also keep a backup of the storage.json file.

**Comment by bxs17** [comment]
miner conf&nbsp;
[Storage]
AllowMove = false?

Is this configuration available
?






------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "filecoin-project/lotus"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年6月28日(星期二) 下午3:34
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [filecoin-project/lotus] After sealing, the file is transferred to HDD. The designated worker works like AP PC1 PC2 tasks enable/disable emove (Discussion #8936)





  
But miner can also do transfer tasks.
  
That is probably because your lotus-miner still has the paths to the long-term storage. The scenario I outlined above, is where the lotus-miner is only doing the scheduling, and has windowPoSt-workers attached, and therefore does not need the paths to the long-term storage on it.
 
In your case it sound like the lotus-miner still has storage-paths set, so if you want to move to an architecture where only the workers does the transfers task you would need to remove them. As a general advice here, please test out this setup first before chaning it on your production. And also keep a backup of the storage.json file.

**Reply by rjan90** [reply]
Hey @bxs17! As far as I know, that is not an configuration available on the `lotus-miner` currently.

**Comment by bxs17** [comment]
So&nbsp; add task configuration in the next version.






&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "filecoin-project/lotus"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年6月30日(星期四) 凌晨3:07
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [filecoin-project/lotus] After sealing, the file is transferred to HDD. The designated worker works like AP PC1 PC2 tasks enable/disable emove (Discussion #8936)





 
Hey @bxs17! As far as I know, that is not an configuration available on the lotus-miner currently.

**Comment by bxs17** [comment]
Which version can use --no-default
Which version can use --no-default


***@***.***

&nbsp;-----------------&nbsp;原始邮件&nbsp;------------------

发件人:                                                                                                                        "filecoin-project/lotus"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年6月28日(星期二) 下午2:51
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [filecoin-project/lotus] After sealing, the file is transferred to HDD. The designated worker works like AP PC1 PC2 tasks enable/disable emove (Discussion #8936)





 
Okay, so if I understand correctly, you want a worker to move the data to the long term storage?
 
If so one option is to just run lotus-worker on storage boxes with all compute tasks disabled; this way all storage accesses are going through independent tcp streams, and straight to whatever filesystem. But that requires:
  
Post workers for reading challenges from remote storage: https://lotus.filecoin.io/storage-providers/seal-workers/post-workers/
  
To run a worker on the a storage box, just so it schedules the moving of sectors, just disable all tasks with: lotus-worker run --no-default

**Comment by TippyFlitsUK** [comment]
Hi @bxs17 

`no-default` is available from release v.1.17.0 onward. The most recent tag is https://github.com/filecoin-project/lotus/releases/tag/v1.17.0-rc2
